ssh.go

```diff
@@ -42,7 +42,57 @@ func validatePubkey(pubkey string, role *sshRole) error {
 	return nil
 }
 
-func validateCert(pubkey ssh.PublicKey, config *ConfigEntry, principals []string) error {
+// Detect the first valid principal from the certificate, if any is present
+func detectPrincipal(pubkey ssh.PublicKey, validPrincipals []string) string {
+	cert, ok := pubkey.(*ssh.Certificate)
+	if !ok {
+		// Not a certificate - no principal to detect
+		return ""
+	}
+
+	if len(validPrincipals) == 0 {
```

Comment:
I think we should make this explicit with specify a `*` in the role.Principals if we want to allow any principal instead of the empty slice.

Empty list should be invalid